native/c/zdsm.c

```diff
@@ -155,12 +155,12 @@ int ZDSRECFM(ZDS *zds, const char *dsn, const char *volser, char *recfm_buf,
     int len = 0;
     char main_fmt = 0;
 
-    if ((dscb->ds1recfm & 0xC0) == 0x40)
+    if ((dscb->ds1recfm & 0xC0) == 0x80)
```

Comment:
is there any doc or comments that you can link to describe why the `& 0x0C` and also where you found that `0x80` is `F` for fixed?

Reply:
It's a part of the PDF already linked in a comment above the function, under section "Format-1 DSCB" - I just mistakenly swapped 0x80 and 0x40. Basically RECFM is a bitmask, so for fixed block it would be: 

`10010000`
and doing a bitmask of 0x80 checks the first byte:

```
    10010000
& 10000000 (0xC0)
---------------------
= 10000000 (0x80) - indicates Fixed
```

I can add a more detailed comment if needed.

Reply:
Sorry, I meant to type `1100 0000` for the binary representation of 0x80. Checking for the presence of the 2nd bit (reading left-to-right) determines whether the data set format is variable.